src/compiler/core.ts

```diff
@@ -2022,7 +2011,7 @@ namespace ts {
         }
 
         const res = previous[s2.length];
-        return res > max ? undefined : res;
+        return res >= max ? undefined : res;

```

Comment:
I guess this is the reason `bestDistance - 1` turned into `bestDistance`? Is it also need to support the switch to Math.max and Math.min, or some other reason, like the switch to non-integral values?

Reply:
😏 Actually, it's the result of `bestDistance - 1` turned into `bestDistance`.  `bestDistance - 1` can be replaced with `bestDistance - 0.1`, but it is very fragile since the minimum of distance could be changed to other values like `0.05` in the future.